crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -424,46 +430,60 @@ async fn trigger_webhook_to_merchant(
                     )
                     .await?;
                 }
+                Some(updated_event)
             }
         },
-    }
+    };
 
-    Ok(())
+    Ok(updated_event)
 }
 
 fn raise_webhooks_analytics_event(
     state: SessionState,
-    trigger_webhook_result: CustomResult<(), errors::WebhooksFlowError>,
+    trigger_webhook_result: CustomResult<Option<domain::Event>, errors::WebhooksFlowError>,
     content: Option<api::OutgoingWebhookContent>,
     merchant_id: common_utils::id_type::MerchantId,
     event: domain::Event,
 ) {
-    let error = if let Err(error) = trigger_webhook_result {
-        logger::error!(?error, "Failed to send webhook to merchant");
+    let mut event = event;
+    let mut error: Option<serde_json::Value> = None;
 
-        serde_json::to_value(error.current_context())
-            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
-            .map_err(|error| {
-                logger::error!(?error, "Failed to serialize outgoing webhook error as JSON");
-                error
-            })
-            .ok()
-    } else {
-        None
-    };
+    match trigger_webhook_result {
+        Ok(updated_event) => {
+            if let Some(updated_event) = updated_event {
+                event = updated_event
+            }
+        }
+        Err(e) => {
+            logger::error!(?error, "Failed to send webhook to merchant");
+
+            error = serde_json::to_value(e.current_context())
+                .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
+                .map_err(|error| {
+                    logger::error!(?error, "Failed to serialize outgoing webhook error as JSON");
+                    error
+                })
+                .ok();
+        }
+    }
 
     let outgoing_webhook_event_content = content
         .as_ref()
         .and_then(api::OutgoingWebhookContent::get_outgoing_webhook_event_content)
         .or_else(|| get_outgoing_webhook_event_content_from_event_metadata(event.metadata));
 
+    // Extract the status_code field from the webhook response
+    let status_code = utils::extract_value_from_response(&event.response, "status_code");
```

Comment:
This need not be a utility function, you can call `event.response.parse_value()` and then access the `status_code` field. You can remove the utility function that was added.

Reply:
Not absolutely necessary, you can call `event.response.peek().parse_value()`.

Reply:
I feel like I'm missing something here. Please clarify. 

In this case `peek()` return `&String` and only available method is `parse_struct()`.
And when attempting to do`event.response.parse_value("status_code")` it requires `ValueExt` trait to be implemented for `Secret<String>`

Reply:
My bad, it's `parse_struct()`.